src/pkg/retention/res/selectors/regexp/selector.go

```diff
@@ -15,32 +15,73 @@
 package regexp
 
 import (
+	"github.com/bmatcuk/doublestar"
 	"github.com/goharbor/harbor/src/pkg/retention/res"
 	"github.com/goharbor/harbor/src/pkg/retention/res/selectors"
 )
 
 const (
 	// Kind ...
 	Kind = "regularExpression"
-	// Matches [pattern]
+	// Matches [pattern] for tag (default)
 	Matches = "matches"
-	// Excludes [pattern]
+	// Excludes [pattern] for tag (default)
 	Excludes = "excludes"
+	// RepoMatches represents repository matches [pattern]
+	RepoMatches = "repoMatches"
+	// RepoExcludes represents repository excludes [pattern]
+	RepoExcludes = "repoExcludes"
 )
 
 // selector for regular expression
 type selector struct {
 	// Pre defined pattern declarator
-	// "matches" and "excludes"
+	// "matches", "excludes", "repoMatches" or "repoExcludes"
 	decoration string
 	// The pattern expression
 	pattern string
 }
 
 // Select candidates by regular expressions
 func (s *selector) Select(artifacts []*res.Candidate) ([]*res.Candidate, error) {
-	// TODO: REPLACE SAMPLE CODE WITH REAL IMPLEMENTATION
-	return artifacts, nil
+	selected := make([]*res.Candidate, 0)
+
+	if len(artifacts) == 0 {
```

Comment:
I don't think you need an explicit check for 0 length here, won't the `for` loop below simply do nothing if the input is empty?